kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/src/main/java/org/kie/workbench/common/dmn/showcase/client/editor/DMNDiagramEditor.java

```diff
@@ -154,7 +156,30 @@ private void initLienzoType() {
         if (canvas != null) {
             LienzoPanel panel = (LienzoPanel) canvas.getView().getPanel();
             LienzoBoundsPanel lienzoPanel = panel.getView();
-            JsCanvas jsCanvas = new JsCanvas(lienzoPanel, lienzoPanel.getLayer());
+            JsCanvas jsCanvas = new JsCanvas(lienzoPanel, lienzoPanel.getLayer(), (UUID, state) -> {
+                Shape shape = stunnerEditor.getCanvasHandler().getCanvas().getShape(UUID);
```

Comment:
@inodeman Considering Both `DMNDiagramEditor` and `BPMNDiagramEditor` share the same `JSShapeStateApplier` logic, is it possible to put this logic in one place (in lienzo client core module), in order to avoid this duplication under the editors' implementations? Thx!

Reply:
Hi @yesamer I tried, but the thing is that method accesses some parts of stunner for instance Shape shape = stunnerEditor.getCanvasHandler().getCanvas().getShape(UUID);, plus it has the states as string, I debated a while what the best way would be and the route with less problems was this since it is only a few lines of codes, I could also create a class in stunner commons but that would only be for a few lines so not sure it is worth it. Let me know what you think

Reply:
@inodeman Considering I don't totally aware of the common hierarchy above DMN Editor and BPMN able, I have no idea where this block of code can be put in order to be used by both editors. My suggestion is still valid, I mean, I think we should avoid this duplication. If this is possible or not, with how much effort etc, I trust you and your analysis, so if you think this is not possible OR the required effort is worthless, it's fine for me. Maybe we can ask for an  @romartin opinion, which is undoubtedly more aware than me about the modules hierarchy here.

Reply:
Hi @yesamer @romartin I have refactored the code, so now that both DMN and BPMN only need to implement an abstract method to return Shape, thus reducing code.